pkg/controller/statusmanager/status_manager.go

```diff
@@ -53,72 +54,76 @@ func New(client client.Client, name, version string) *StatusManager {
 
 // Set updates the ClusterOperator.Status with the provided conditions
 func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...configv1.ClusterOperatorStatusCondition) {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
+		err := status.client.Get(context.TODO(), types.NamespacedName{Name: status.name}, co)
+		isNotFound := errors.IsNotFound(err)
+		if err != nil && !isNotFound {
+			log.Printf("Failed to get ClusterOperator %q: %v", status.name, err)
+			return err
```

Comment:
you're logging `err` outside of the `RetryOnConflict`, so if you log here too we're going to get double error messages. I think you can just remove all the `log.Printf` calls inside the loop. The single error message outside the loop is fine.

Reply:
I thought it would be good to get a lower layer log in case something fails - i.e if it's the `Get`,`Create`,`Update`. Besides, today we log `Updated ClusterOperator with conditions` for every successful update, is removing that acceptable?

Reply:
Definitely keep the non-error logging ("Updated/Created ClusterOperator", etc).

For the errors, they shouldn't really happen, and I think there should be enough information to figure them out without needing further clarification anyway. eg, if it's "Not Found" then that's obviously an error on the "Get". And if it's "no route to host" or "timed out", it doesn't really matter which operation it occurred on...

Reply:
Ok, just updated the PR taking into account your input.